### PR TITLE
Draft proposals v2

### PR DESCRIPTION
**Pull request by mcgingras**

### CODE REVIEW

**Comment by jefag** [comment]
We need a passing build, before we give to Bryan.

**Comment by mcgingras** [comment]
## Issues
- Creating a public draft with no sponsors does not work

File: src/app/api/common/proposals/getProposals.ts [hunk 1]
```diff
@@ -164,9 +175,223 @@ async function getTotalProposalsCount(): Promise<number> {
   });
 }
 
+const getDraftProposalsV2 = async (
```
Comment (mcgingras) [inline src/app/api/common/proposals/getProposals.ts:178]: maybe should change this back to just `getDraftProposals` after deprecating the other

**Comment by mcgingras** [comment]
@jefag @CarlGocht @pedrohasf @sudheerDev 

I rebased these changes off of main locally. I thought it might be helpful for me to keep this branch rebased periodically but I want to make sure it's okay with you guys before I push up changes. Honestly, I'm not even sure if I have write access anymore, but before I tested that out I wanted to run it by y'all before you got a mystery commit to this branch. 

Cheers y'all, let me know.

**Comment by jefag** [comment]
Hey @mcgingras , that's generous of you. Thank you for thinking of us and helping.

I think taking an open-source approach from here makes sense.  With that as our guide, you can safely assume you own this branch, until its merged.  Somebody internally is going to pick this up, but until they push what they have.  Whatever they are or aren't doing on their local, is their problem until they socialize.  

A rebase would be helpful.  

tl;dr feel free to force-push!

PS - I believe you still have rights to "write" this repo.

**Comment by mcgingras** [comment]
Thanks @jefag! Sorry, I didn't see your notification sooner. I need to be more on top of github notifications :-0

**Comment by mcgingras** [comment]
I don't have much context on [this PR](https://github.com/voteagora/agora-next/pull/857) but my guess is we will need these changes into the drafts v2 branch as well, assuming these are changes that affect the way we present proposal creation to users.

**Comment by sudheerDev** [comment]
> I don't have much context on [this PR](https://github.com/voteagora/agora-next/pull/857) but my guess is we will need these changes into the drafts v2 branch as well, assuming these are changes that affect the way we present proposal creation to users.

Hey @mcgingras the change was just for OP. I can make the same changes on top of this branch before we merge the PR. That should be fine

**Comment by mcgingras** [comment]
thanks for reopening Jeff, I check on my baby every so often
